test/all-functional.spec.ts

```diff
@@ -14,6 +14,7 @@ const IGNORED_MODULES = new Set([
 function getMethodNamesByModules(faker: Faker): { [module: string]: string[] } {
   return Object.fromEntries(
     Object.keys(faker)
+      // eslint-disable-next-line unicorn/no-array-callback-reference -- we need this for type narrowing
       .filter(isTestableModule)
```

Comment:
Does moving the type guard to the inline return type not work with this rule? I don't see why this should be an exception if the rule doesn't allow any.

```suggestion
      .filter((v): v is keyof Faker => isTestableModule(v))
```